### PR TITLE
Create pro-mode design and add camera panel (gallery, take photo, face mode switch)

### PR DESCRIPTION
**Pull request by aleksazh**
Completed tasks:
1. Refactor code of Pro demo.
2. Create a new design for Pro demo.
3. Add camera panel with the following functionality:
   * Gallery (only canvas with the last photo as of now)
   * Take photo
   * Face mode button (switch camera to user or environment mode)

### CODE REVIEW

File: samples/camera/index.html [hunk 1]
```diff
@@ -5,68 +5,98 @@
   <meta charset="UTF-8">
   <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
   <title>Image Capture Capabilities</title>
-  <link rel="stylesheet" href="../css/style.css">
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (riju) [inline samples/camera/index.html]: I understand this is the recommended way, but we need to account for offline cases also .. Just add a note saying that we will look into self-hosting.

File: samples/camera/js/ui.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    videoTrack.applyConstraints(constraint).catch(e => console.log(e));
+  });
+
+  // TakePhoto event by clicking takePhotoButton.
+  let takePhotoButton = document.getElementById('takePhotoButton');
+  takePhotoButton.addEventListener('click', function () {
+    takePhoto();
+    const canvas = document.getElementById('gallery');
+    canvas.getContext('2d').rotate(-90 * Math.PI / 180); // rotate 90 degrees
```
Comment (riju) [inline samples/camera/js/ui.js]: Add a note here why you are rotating.

File: samples/camera/js/ui.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  videoTrack = video.srcObject.getVideoTracks()[0];
+  imageCapturer = new ImageCapture(videoTrack);
+  // Timeout needed in Chrome, see https://crbug.com/711524.
+  setTimeout(() => {
+    applyInitialSettings();
+  }, 500);
+}
+
+function displayColorTemperatureIcon() {
+  let id = 'colorTemperature';
```
Comment (riju) [inline samples/camera/js/ui.js]: Add this Link here https://w3c.github.io/mediacapture-image/#white-balance-mode

File: samples/camera/js/ui.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  if (controls[id].value >= 2500 && controls[id].value <= 3500)
```
Comment (riju) [inline samples/camera/js/ui.js]: use a switch case here ?

```js

let text =""
switch(controls[id].value)
case (> 9000): 
    break
case (> 8000):
   text = 'brightness_3'; break;
..
document.getElementById('wbIcon').innerText ='wb_' + text
```
I just checked [this](http://jsfiddle.net/some/HKdug/), where its asking to use if-then or switch with immediate values and not range

We have work with ranges others wise map could have been a good choice
```js
let colorTemperature = {
  9000 : '',
  8000 : "twilight",
  6500 : "cloudy",
  ...
};
```

document.getElementById('wbIcon').innerText = Object.keys(controls[id].value)

File: samples/faceDetection/index.js [hunk 5]
```diff
@@ -117,17 +113,51 @@ function startCamera() {
...
+
+  controls = {
+    frontCamera: null,
+    backCamera: null,
+    facingMode: '',
+  };
+
+  let facingModeButton = document.getElementById('facingModeButton');
+  // Switch to face or environment mode by clicking facingModeButton.
+  facingModeButton.addEventListener('click', function () {
```
Comment (riju) [inline samples/faceDetection/index.js]: Can this be moved to Util ? I think all the demos are using this right ?

File: samples/funnyHats/js/ui.js [hunk 6]
```diff
@@ -103,26 +100,37 @@ function initUI() {
...
+    drawCanvas(dstCanvas, canvasOutput);
+  });
+
+  controls = {
+    frontCamera: null,
+    backCamera: null,
+    facingMode: '',
+  };
+
+  let facingModeButton = document.getElementById('facingModeButton');
```
Comment (riju) [inline samples/funnyHats/js/ui.js]: move to utils ?

**Comment by aleksazh** [comment]
Should I squash `0335a65` and `004ffa6` commits?

**Comment by riju** [comment]
Ya .. squash all :)

**Comment by aleksazh** [comment]
Done

**Comment by riju** [comment]
👏👏

**Pull request merged**